libgadget/utils/system.c

```diff
@@ -609,33 +584,29 @@ MPIU_write_pids(char * filename)
     int ThisTask;
     MPI_Comm_size(comm, &NTask);
     MPI_Comm_rank(comm, &ThisTask);
-    pid_t my_pid;
-    char mode[8], buf[500];
-    FILE *fd;
-    int i;
-
-    my_pid = getpid();
-
-    strcpy(mode, "a+");
 
-    for(i = 0; i < NTask; i++)
+    int my_pid = getpid();
+    int * pids = ta_malloc("pids", int, NTask);
+    char * hosts = ta_malloc("hosts", char, NTask * 32);
+    char host[32]={"NULL"};
+    char * genv = getenv("HOST");
```

Comment:
Use get_hostname rather than HOST.

Also the max size limit here is 32, but the one above was (bufsz) 256. I think 64 is a pretty safe value -- perhaps use that instead?

Reply:
Oh nice catch! I'll change the pidfile to match cluster_get_num_hosts. 

in cluster_get_num_hosts, the extra buffer size costs us nothing and could potentially help us some day when someone runs on a cluster with really long node names. But in pidfile an overflow is benign but running out of memory is a potential issue.